eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/FieldEmpty.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+/**
+ * File containing the DoctrineDatabase field criterion handler class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Search\Legacy\Content\Common\Gateway\CriterionHandler;
+
+use eZ\Publish\API\Repository\FieldTypeService;
+use eZ\Publish\Core\Search\Legacy\Content\Common\Gateway\CriteriaConverter;
+use eZ\Publish\API\Repository\Exceptions\NotImplementedException;
+use eZ\Publish\Core\Persistence\Database\DatabaseHandler;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\Core\Persistence\Legacy\Content\FieldValue\ConverterRegistry as Registry;
+use eZ\Publish\Core\Base\Exceptions\InvalidArgumentException;
+use eZ\Publish\Core\Persistence\Database\SelectQuery;
+use eZ\Publish\SPI\Persistence\Content\Type\Handler as ContentTypeHandler;
+use eZ\Publish\SPI\Persistence\Content\Language\Handler as LanguageHandler;
+
+/**
+ * Field criterion handler.
+ */
+class FieldEmpty extends FieldBase
+{
+    /**
+     * Field converter registry.
+     *
+     * @var \eZ\Publish\Core\Persistence\Legacy\Content\FieldValue\ConverterRegistry
+     */
+    protected $fieldConverterRegistry;
+
+    /**
+     * @var \eZ\Publish\API\Repository\FieldTypeService
+     */
+    protected $fieldTypeService;
+
+    /**
+     * Construct from handler handler.
```

Comment:
What happened here? ;)
All types are strict, so you can skip this phpdoc anyway.

Reply:
Bad case of copy&pase :slightly_smiling_face: 
https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/Field.php#L50